website/blog/2022-07-26-configuring-grants.md

```diff
@@ -0,0 +1,211 @@
+
+#### The solution today
+
+dbt’s new `grants` implementation takes account of this. It knows whether grants are “carried over” when a model is re-run based on its materialization and your database. It makes up the difference between the existing grants and the ones you actually want.
+
+Try it out!
+
+```sql
+
+-- models/my_incremental_model.sql
+
+{{ config(
+	materialized = 'incremental',
+	grants = {'select': ['another_user']}
+) }}
+
+select ...
+```
+
+Run that, verify that `another_user` can select from your model. Then change your model and run it again:
+
+```sql
+-- models/my_incremental_model.sql
+
+{{ config(
+	materialized = 'incremental',
+	grants = {'select': []}
+) }}
+
+select ...
+```
+
+If you check your database, you should see that *no one* can select from the incremental model. You could also see, in the debug-level logs, that dbt has run a `revoke` statement.
+
+(Note that, if `grants` is missing or set to `{}`, dbt will understand that you don’t want it managing grants for this table. So it’s best to explicitly specify the privilege, and that you want *no one* to have it!)
+
+Great! Now that you’re using the `grants` feature in dbt v1.2, you’ve just given this more thought than you should ever need to again 😎
+
+## Is there still a place for hooks?
+
+Yes, indeed! Some areas that stand out:
+- [Granting permissions on other object types](#granting-permissions-on-other-object-types) like granting usage on a schema
+- [Advanced permissions](#advanced-permissions-or-other-operations) like row-level access
+
+### Granting permissions on other object types
+
+For now, it’s still necessary to grant `usage` on schemas to users that will need to select from objects in those schemas. Even though dbt creates schemas at the start of runs, there isn’t really a way to configure *schemas as their own objects* within dbt.
+
+Here's a couple ways you could approach it:
+- Option A -- simple and familiar -- hooks to the rescue
+- Option B -- too clever by half -- use the dbt graph to infer which schemas need "usage"
+
+Both examples: [https://github.com/dbt-labs/internal-analytics/compare/jerco/refactor-grants](https://github.com/dbt-labs/internal-analytics/compare/jerco/refactor-grants)
+
+#### Option A: simple and familiar
+
+```yaml
+on-run-end:
+	# better as a macro
+	- "{% for schema in schemas %}grant usage on schema {{ schema }} to reporter;{% endfor %}"
+```
+
+Upside: Short, sweet, to the point.
+
+Downside: we need to repeat the same list of roles here that we specified in our `grants` config.
+
+#### Option B: Too clever by half
+
+Now that `grants` is a real config in dbt, available via dbt metadata, you can do all sorts of fun things with it. For instance, figure out which schemas have at least one object granting `select` to a role, and then grant `usage` on that schema to that role!
+
+```sql
+-- macros/operations/reporting_grants.sql
+{% macro grant_usage_on_schemas_where_select() %}
+    /*
+      TOTALLY UNTESTED!!!
```

Comment:
Let's possibly remove this line 😅 

Could say: "Pseudo code only, for demonstration purposes"